vulnerabilities/importers/mozilla.py

```diff
@@ -0,0 +1,208 @@
+
+        advisories = []
+        for path in files:
+            for advisory in self._to_advisories(path):
+                advisories.append(advisory)
+                if len(advisories) >= self.batch_size:
+                    yield advisories
+                    advisories = []
+
+        # In case batch size is too high
+        yield advisories
+
+    def _to_advisories(self, path: str) -> List[Advisory]:
+        """
+        Convert a file to corresponding advisories.
+        This calls proper method to handle yml/md files.
+        """
+        mfsa_id = self._mfsa_id_from_filename(path)
+
+        with open(path) as lines:
+            if path.endswith(".md"):
+                return self._parse_md(mfsa_id, lines)
+            elif path.endswith(".yml"):
+                return self._parse_yml(mfsa_id, lines)
+
+        return []
+
+    def _parse_yml(self, mfsa_id, lines) -> List[Advisory]:
+        advisories = []
+        data = yaml.safe_load(lines)
+        data["mfsa_id"] = mfsa_id
+
+        fixed_package_urls = self._get_package_urls(data.get("fixed_in"))
+        references = self._get_references(data)
+
+        if not data.get("advisories"):
+            return []
+
+        for cve, advisory in data["advisories"].items():
+            advisories.append(
+                Advisory(
+                    summary=advisory.get("description"),
+                    vulnerability_id=cve if is_cve(cve) else "",
+                    impacted_package_urls=[],
+                    resolved_package_urls=fixed_package_urls,
+                    references=references,
+                )
+            )
+
+        return advisories
+
+    def _parse_md(self, mfsa_id, lines) -> List[Advisory]:
+        yamltext, mdtext = self._parse_md_front_matter(lines)
+
+        data = yaml.safe_load(yamltext)
+        data["mfsa_id"] = mfsa_id
+
+        fixed_package_urls = self._get_package_urls(data.get("fixed_in"))
+        references = self._get_references(data)
+
+        description = self._html_get_p_under_h3(markdown(mdtext), "description")
+
+        # FIXME: add references from md ? They lack a proper reference id and are mostly bug reports
+
+        return [
+            Advisory(
+                summary=description,
+                vulnerability_id="",  # FIXME: Scrape the entire page for CVE regex ?
+                impacted_package_urls=[],
+                resolved_package_urls=fixed_package_urls,
+                references=references,
+            )
+        ]
+
+    def _html_get_p_under_h3(self, html, h3: str):
+        soup = BeautifulSoup(html, features="lxml")
+        h3tag = soup.find("h3", text=lambda txt: txt.lower() == h3)
+        p = ""
+        if h3tag:
+            for tag in h3tag.next_siblings:
+                if tag.name:
+                    if tag.name != "p":
+                        break
+                    p += tag.get_text()
+        return p
+
+    def _parse_md_front_matter(self, lines):
```

Comment:
I guess you meant https://github.com/nexB/vulnerablecode/blob/3d66b4e82ee31422ab907d3388f739b768ffd2ac/vulnerabilities/importers/istio.py#L86
Yes. It would make more sense as a helper. Should I name it `extract_markdown_front_matter` or let me know if you have something  better in mind.